### PR TITLE
Remove usage of v1 api for CloudWatchLogs, CloudTrail and ASG and replace it with v2

### PR DESCRIPTION
**Pull request by aclevername**
### Description

Closes https://github.com/weaveworks/eksctl/issues/4879. Removes the usage of aws-sdk-go v1 for CloudWatchLogs, CloudTrail and ASG and replace it with aws-sdk-go-v2

### CODE REVIEW

File: pkg/actions/nodegroup/scale.go [hunk 1]
```diff
@@ -66,19 +67,19 @@ func (m *Manager) scaleUnmanagedNodeGroup(ng *api.NodeGroupBase, stackInfo manag
...
 
 	if ng.DesiredCapacity != nil {
-		input.DesiredCapacity = aws.Int64(int64(*ng.DesiredCapacity))
+		input.DesiredCapacity = aws.Int32(int32(*ng.DesiredCapacity))
 	}
-	out, err := m.ctl.Provider.ASG().UpdateAutoScalingGroup(input)
+
+	_, err := m.ctl.Provider.ASG().UpdateAutoScalingGroup(ctx, input)
 	if err != nil {
-		logger.Debug("ASG update output: %s", out.String())
```
Comment (Skarlso) [inline pkg/actions/nodegroup/scale.go:81]: We don't need this output anymore? 🤔

**Review by Skarlso — APPROVED** [review]
LGTM with one question

File: pkg/actions/nodegroup/scale.go (hunk 1 above)
Comment (aclevername) [reply pkg/actions/nodegroup/scale.go:81]: `out.String()` is no longer a thing, so I wasn't really sure if it made sense to just print th estruct. I'm not really sure, but given its a debug and not info output i think its okay to just remove :shrug: wdyt?

Comment (Skarlso) [reply pkg/actions/nodegroup/scale.go:81]: nah

**Pull request merged**